src/ruby/lib/grpc/errors.rb

```diff
@@ -42,12 +43,28 @@ def initialize(code, details = 'unknown cause', metadata = {})
       @metadata = metadata
     end
 
-    # Converts the exception to a GRPC::Status for use in the networking
+    # Converts the exception to a {Struct::Status} for use in the networking
     # wrapper layer.
     #
-    # @return [Status] with the same code and details
+    # @return [Struct::Status] with the same code and details
     def to_status
-      Struct::Status.new(code, details, @metadata)
+      Struct::Status.new(code, details, metadata)
+    end
+
+    # Converts the exception to a deserialized {Google::Rpc::Status} object.
+    # Returns `nil` if the `grpc-status-details-bin` trailer could not be
+    # converted to a {Google::Rpc::Status} due to the server not providing
+    # the necessary trailers.
+    #
+    # @return [Google::Rpc::Status, nil]
+    def to_rpc_status
+      status = to_status
+
+      return if status.nil?
+
+      GoogleRpcStatusUtils.extract_google_rpc_status(status)
+    rescue
```

Comment:
Add a `GRPC.logger.warn` in this rescue block?

Reply:
Added.